integration-tests/testconfig/automation/config.go

```diff
@@ -140,3 +148,197 @@ func (c *DataStreams) Validate() error {
 	}
 	return nil
 }
+
+type AutomationConfig struct {
+	PluginConfig     *PluginConfig     `toml:"PluginConfig"`
+	PublicConfig     *PublicConfig     `toml:"PublicConfig"`
+	RegistrySettings *RegistrySettings `toml:"RegistrySettings"`
+	UseLogBufferV1   *bool             `toml:"use_log_buffer_v1"`
+}
+
+func (c *AutomationConfig) Validate() error {
+	if err := c.PluginConfig.Validate(); err != nil {
+		return err
+	}
+	if err := c.PublicConfig.Validate(); err != nil {
+		return err
+	}
+	if err := c.RegistrySettings.Validate(); err != nil {
+		return err
+	}
+	if c.UseLogBufferV1 == nil {
+		return errors.New("use_log_buffer_v1 must be set")
+	}
+	return nil
+}
+
+type PluginConfig struct {
+	PerformLockoutWindow *int64             `toml:"perform_lockout_window"`
+	TargetProbability    *string            `toml:"target_probability"`
+	TargetInRounds       *int               `toml:"target_in_rounds"`
+	MinConfirmations     *int               `toml:"min_confirmations"`
+	GasLimitPerReport    *uint32            `toml:"gas_limit_per_report"`
+	GasOverheadPerUpkeep *uint32            `toml:"gas_overhead_per_upkeep"`
+	MaxUpkeepBatchSize   *int               `toml:"max_upkeep_batch_size"`
+	LogProviderConfig    *LogProviderConfig `toml:"LogProviderConfig"`
+}
+
+type LogProviderConfig struct {
+	BlockRate *uint32 `toml:"block_rate"`
+	LogLimit  *uint32 `toml:"log_limit"`
+}
+
+func (c *PluginConfig) Validate() error {
+	if err := c.LogProviderConfig.Validate(); err != nil {
+		return err
+	}
+	if *c.PerformLockoutWindow < 0 || c.PerformLockoutWindow == nil {
```

Comment:
for this one and all the following ones, you should check for `nil` first, otherwise it will panic if it's `nil` and you try to indirect it, no?